src/ghcup/yaml.rs

```diff
@@ -33,11 +36,30 @@ impl SnapshotStorage<SnapshotMeta> for GhcupYaml {
 
         info!(logger, "fetching ghcup config...");
         progress.set_message("downloading version file");
-        let yaml_url = get_yaml_url(base_url, &client).await?;
+        let yaml_url = Url::parse(get_yaml_url(base_url, &client).await?.as_str())
+            .map_err(|_| Error::ProcessError(String::from("invalid ghcup yaml url")))?;
+
+        // additional yaml paths
+        let mut yaml_paths = self
+            .additional_yaml
+            .iter()
+            .map(|filename| {
+                let mut new_yaml_url = yaml_url.clone();
+                {
+                    let mut segments = new_yaml_url
+                        .path_segments_mut()
+                        .map_err(|_| Error::ProcessError(String::from("invalid ghcup yaml url")))?;
+                    segments.pop().push(filename);
+                }
+                Ok(new_yaml_url.path()[1..].to_string())
```

Comment:
Why ignore the first element? Please add comments for this.

Reply:
remove first char (slash), comment added